[
  {
    "package": {
      "name": "idna",
      "version": "3.6",
      "ecosystem": "PyPI"
    },
    "dependency_groups": [
      "prowler-requirements"
    ],
    "vulnerabilities": [
      {
        "modified": "2024-04-11T21:56:51Z",
        "published": "2024-04-11T21:32:40Z",
        "schema_version": "1.6.0",
        "id": "GHSA-jjg7-2v4v-x38h",
        "aliases": [
          "CVE-2024-3651"
        ],
        "summary": "Internationalized Domain Names in Applications (IDNA) vulnerable to denial of service from specially crafted inputs to idna.encode",
        "details": "### Impact\nA specially crafted argument to the `idna.encode()` function could consume significant resources. This may lead to a denial-of-service.\n\n### Patches\nThe function has been refined to reject such strings without the associated resource consumption in version 3.7.\n\n### Workarounds\nDomain names cannot exceed 253 characters in length, if this length limit is enforced prior to passing the domain to the `idna.encode()` function it should no longer consume significant resources. This is triggered by arbitrarily large inputs that would not occur in normal usage, but may be passed to the library assuming there is no preliminary input validation by the higher-level application.\n\n### References\n* https://huntr.com/bounties/93d78d07-d791-4b39-a845-cbfabc44aadb",
        "affected": [
          {
            "package": {
              "ecosystem": "PyPI",
              "name": "idna",
              "purl": "pkg:pypi/idna"
            },
            "ranges": [
              {
                "type": "ECOSYSTEM",
                "events": [
                  {
                    "introduced": "0"
                  },
                  {
                    "fixed": "3.7"
                  }
                ]
              }
            ],
            "versions": [
              "0.1",
              "0.2",
              "0.3",
              "0.4",
              "0.5",
              "0.6",
              "0.7",
              "0.8",
              "0.9",
              "1.0",
              "1.1",
              "2.0",
              "2.1",
              "2.10",
              "2.2",
              "2.3",
              "2.4",
              "2.5",
              "2.6",
              "2.7",
              "2.8",
              "2.9",
              "3.0",
              "3.1",
              "3.2",
              "3.3",
              "3.4",
              "3.5",
              "3.6"
            ],
            "database_specific": {
              "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2024/04/GHSA-jjg7-2v4v-x38h/GHSA-jjg7-2v4v-x38h.json"
            }
          }
        ],
        "severity": [
          {
            "type": "CVSS_V3",
            "score": "CVSS:3.1/AV:L/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H"
          }
        ],
        "references": [
          {
            "type": "WEB",
            "url": "https://github.com/kjd/idna/security/advisories/GHSA-jjg7-2v4v-x38h"
          },
          {
            "type": "PACKAGE",
            "url": "https://github.com/kjd/idna"
          }
        ],
        "database_specific": {
          "cwe_ids": [
            "CWE-400"
          ],
          "github_reviewed": true,
          "github_reviewed_at": "2024-04-11T21:32:40Z",
          "nvd_published_at": null,
          "severity": "MODERATE"
        }
      }
    ],
    "groups": [
      {
        "ids": [
          "GHSA-jjg7-2v4v-x38h"
        ],
        "aliases": [
          "CVE-2024-3651",
          "GHSA-jjg7-2v4v-x38h"
        ],
        "max_severity": "6.2"
      }
    ]
  }
]